### DISCUSSION TITLE
BlenderBIM Add-on: Construction Scheduling

### DISCUSSION THREAD

**Discussion by MahdiA2024** [IfcOpenShell/IfcOpenShell#4770] [Q&A]
Hi
![time 2](https://github.com/IfcOpenShell/IfcOpenShell/assets/171282714/f930e528-897f-47f0-8d2c-36a375f625a5)
![time 1](https://github.com/IfcOpenShell/IfcOpenShell/assets/171282714/9e0b4e8a-e7a3-4ec3-a892-e124077f94f2)

In the Construction Scheduling tab, I created Work Calendars. I added a weekly time schedule for Monday, Tuesday, Wednesday, Thursday, and Friday.

I started adding a few tasks and created a timeline for the tasks:

Schedule Start: 01.05.2024
Schedule Finish: 08.05.2024
Schedule Duration: 8 days

The duration is shown as 8 days, but the correct duration should be 6 days because my working week excludes weekends (i.e., Saturday and Sunday). This means two days should be subtracted from the 8 days, but this adjustment has not happened. Can anyone help me? I want to know the solution for calculating the duration without weekends. Is this duration correct, or is there a programming problem or an unknown solution?

**Comment by Moult** [comment]
Have you applied the calendar to your tasks? Generally you should apply the calendar to the top level task, and then all subtasks will inherit it (I.e. you will set a default calendar)

**Reply by MahdiA2024** [reply]
Thank you for help. I thought my calendar was automatically applied, but now it works after I applied it manually.

**Comment by theoryshaw** [comment]
Potential answer here: https://community.osarch.org/discussion/comment/20802/#Comment_20802
